src/context/context.ts

```diff
@@ -278,21 +278,50 @@ export class WorkflowContext<TInitialPayload = unknown> {
    * @returns call result (parsed as JSON if possible)
    */
   // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-parameters
-  public async call<TResult = unknown, TBody = unknown>(
+  public async call(
     stepName: string,
-    url: string,
-    method: HTTPMethods,
-    body?: TBody,
-    headers?: Record<string, string>
-  ) {
+    callSettings: {
+      url: string;
+      method?: HTTPMethods;
+      body?: unknown;
+      headers?: Record<string, string>;
+    }
+  ): Promise<CallResponse> {
+    const { url, method = "GET", body, headers = {} } = callSettings;
+
     const result = await this.addStep(
-      new LazyCallStep<string>(stepName, url, method, body, headers ?? {})
+      new LazyCallStep<CallResponse | string>(stepName, url, method, body, headers ?? {})
     );
 
+    // <for backwards compatibity>
+    // if you transition to upstash/workflow from upstash/qstash,
+    // the out field in the steps will be the body of the response.
+    // we need to handle them explicitly here
+    if (typeof result === "string") {
+      try {
+        const body = JSON.parse(result);
+        return {
+          status: -1,
```

Comment:
Why do we return status `-1` on succesful request?

Reply:
it could be anything from 200 to 299, so I didn't want to return anything, but let's go with 200.